workers/sentry/src/index.ts

```diff
@@ -0,0 +1,134 @@
+   */
+  public type: string = pkg.workerType;
+
+  /**
+   * Message handle function
+   *
+   * @param event - event to handle
+   */
+  public async handle(event: SentryEventWorkerTask): Promise<void> {
+    /**
+     * Define  event type
+     */
+    this.type = 'errors/sentry';
+
+    const rawEvent = b64decode(event.payload.envelope);
+    const envelope = this.parseSentryEnvelope(rawEvent);
+    this.logger.debug(JSON.stringify(envelope));
+
+    // Todo: For now, we only handle the first item in the envelope
+    const hawkEvent = this.transformToHawkFormat(envelope.Header, envelope.Items[0], event.projectId);
+    this.logger.debug(JSON.stringify(hawkEvent));
+
+    this.validate(hawkEvent);
+
+    await this.addTask(WorkerNames.DEFAULT, hawkEvent as DefaultEventWorkerTask);
+  }
+
+  /**
+    * Parse Sentry envelope into structured format
+    *
+    * @param data - raw Sentry envelope data
+    * @returns Parsed SentryEnvelope object
+    */
+  private parseSentryEnvelope(data: string): SentryEnvelope {
+    const lines = data.split('\n').filter(line => line.trim() !== '');
+    const envelope: SentryEnvelope = { Header: {}, Items: [] };
+
+    // Parse envelope header
+    const headerLine = lines.shift();
+    if (!headerLine) {
+      throw new Error('Failed to read envelope header');
+    }
+    envelope.Header = JSON.parse(headerLine);
+
+    // Parse each item
+    while (lines.length > 0) {
+      // Item header
+      const itemHeaderLine = lines.shift();
+      if (!itemHeaderLine) {
+        throw new Error('Failed to read item header');
+      }
+      const itemHeader = JSON.parse(itemHeaderLine);
+
+      // Item payload
+      const itemPayloadLine = lines.shift();
+      if (!itemPayloadLine) {
+        throw new Error('Failed to read item payload');
+      }
+      const itemPayload = JSON.parse(itemPayloadLine);
+
+      envelope.Items.push({ Header: itemHeader, Payload: itemPayload });
+    }
+
+    return envelope;
+  }
+
+  /**
+   * Transform a Sentry event into a Hawk-compatible structure
+   *
+   * @param envelopeHeader - Sentry envelope header
+   * @param eventItem - Sentry event item
+   * @returns Hawk-compatible event structure
+   */
+  private transformToHawkFormat(
+    envelopeHeader: Record<string, any>,
+    eventItem: SentryItem,
+    projectId: string
+  ): DefaultEventWorkerTask {
+    const { sent_at, trace } = envelopeHeader;
+    const { Payload } = eventItem;
+
+    // delete public_key from trace
+    delete trace.public_key;
+
+    // convert sent_at from ISO 8601 to Unix timestamp
+    const sent_at_unix = Math.floor(new Date(sent_at).getTime() / 1000);
+
+    const backtrace = Payload.exception?.values?.[0]?.stacktrace?.frames?.map((frame: any) => ({
+      file: frame.filename,
+      line: frame.lineno,
+      function: frame.function,
+      sourceCode: frame.pre_context?.concat(frame.context_line, frame.post_context || [])
```

Comment:
why there are only 5 rows above? where is the 5 rows below? I think, sentry should send them as well

<img width="804" alt="image" src="https://github.com/user-attachments/assets/7fc4b901-c0ba-4fcb-a398-66a3cfd068f2" />

Reply:
It's okay. This just the last line of the script :) if you run script with error on not the last row – there will be 5 rows after
<img width="864" alt="image" src="https://github.com/user-attachments/assets/ede3b4d2-2adb-4acd-a648-42e985ae4339" />